builder/vsphere/driver/vm_test.go

```diff
@@ -60,3 +60,46 @@ func TestVirtualMachineDriver_Configure(t *testing.T) {
 		t.Fatalf("Configure should fail")
 	}
 }
+
+func TestVirtualMachineDriver_CreateVM(t *testing.T) {
```

Comment:
I wasn't able to test this with a real environment yet but this test, which uses a simulator, fails for the same reported issue without the changes from this PR. With the changes, it starts passing 👍🏼

Reply:
Before the change: 
```
=== RUN   TestVirtualMachineDriver_CreateVM
    vm_test.go:103: unexpected error *types.InvalidDeviceSpec
--- FAIL: TestVirtualMachineDriver_CreateVM (0.04s)
```

After the change: 
```
=== RUN   TestVirtualMachineDriver_CreateVM
--- PASS: TestVirtualMachineDriver_CreateVM (0.04s)
PASS
```